### PR TITLE
Formatter incorrectly making Relationfields Optional

### PR DESCRIPTION
**Pull request by do4gr**
Fixes https://github.com/prisma/prisma/issues/3548

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -35,9 +35,13 @@ members = [
   "libs/user-facing-errors",
 ]
 
+[profile.dev]
+split-debuginfo = "unpacked"
```
Comment (tomhoule) [inline Cargo.toml:39]: This is a bit obscure if you don't have context. Can we make it a separate commit with a message that links to the blog post for explanation? (or maybe a comment)

File: libs/datamodel/core/src/transform/ast_to_dml/standardise_formatting.rs [hunk 2]
```diff
@@ -257,6 +260,8 @@ impl StandardiserForFormatting {
                                 }
                                 Some(other) if other.field_type().is_compatible_with(&f.field_type) => {
                                     // field with name exists and its type is compatible. We must not add it since we would have a duplicate.
+                                    //todo collect arities of fields on other model
+                                    if other.arity().is_optional() {all_existing_underlying_fields_on_opposite_model_are_required =false;}
```
Comment (tomhoule) [inline libs/datamodel/core/src/transform/ast_to_dml/standardise_formatting.rs:264]: we'll revisit that soon (hopefully)

**Review by tomhoule — APPROVED** [review]

File: Cargo.toml (hunk 1 above)
Comment (do4gr) [reply Cargo.toml:39]: Yep, will do
